From: Ido Schimmel <idosch@nvidia.com>
Date: Wed, 26 May 2021 21:00:04 +0300
Subject: [PATCH] Fix compilation error with gcc 11
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix following compilation error:

lookup_bfd.c: In function ‘lookup_bfd_init’:
lookup_bfd.c:39:54: error: ‘sprintf’ may write a terminating nul past the end of the destination [-Werror=format-overflow=]
   39 |         sprintf(dbibuf,"/usr/lib/debug/lib/modules/%s", uts.release);
      |                                                      ^
lookup_bfd.c:39:9: note: ‘sprintf’ output between 28 and 287 bytes into a destination of size 286
   39 |         sprintf(dbibuf,"/usr/lib/debug/lib/modules/%s", uts.release);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Ido Schimmel <idosch@nvidia.com>
---
 src/lookup_bfd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lookup_bfd.c b/src/lookup_bfd.c
index 8fa4833d8070..e09dc3f2ddc0 100644
--- a/src/lookup_bfd.c
+++ b/src/lookup_bfd.c
@@ -35,7 +35,7 @@ static int lookup_bfd_init(void)
 	if(uname(&uts)<0)
 		return-1;
 
-	dbibuf = malloc(strlen("/usr/lib/debug/lib/modules") + strlen(uts.release) + 1);
+	dbibuf = malloc(strlen("/usr/lib/debug/lib/modules/") + strlen(uts.release) + 1);
 	sprintf(dbibuf,"/usr/lib/debug/lib/modules/%s", uts.release);
 	if (stat(dbibuf,&sb) < 0) {
 		free(dbibuf);
